doc/howto/write_a_view/index.md

```diff
@@ -0,0 +1,429 @@
+    a few more member functions, but it's good practice and will help you understand the concept.
+  * The `static_assert` will just return `true` or `false`, to get more detailed information on why your type does
+    not (yet) model the concept, implement a constrained function template and pass an object of your type to that
+    template.
+  * Your iterator needs to be comparable to the sentinel of the underlying range, you can access that type via
+    `std::ranges::sentinel_t<urng_t>`.
+
+\endassignment
+
+\solution
+\snippet doc/howto/write_a_view/solution_iterator.cpp start
+\snippet doc/howto/write_a_view/solution_iterator.cpp solution1a
+```cpp
+    using reference             = typename std::iterator_traits<base_t>::reference;
+
+```
+\snippet doc/howto/write_a_view/solution_iterator.cpp solution1b
+\snippet doc/howto/write_a_view/solution_iterator.cpp end
+
+The program prints "G A T T A C A ".
+\endsolution
+
+\assignment{Exercise 4: A transforming iterator}
+
+In the previous assigment you have created a working, but pointless iterator.
+It does not do anything differently from the original.
+
+Your task now is to implement the "complementing" behaviour, i.e.
+  * your iterator should only work on ranges of `seqan3::NucleotideAlphabet`
+  * when accessing an element through the iterator it should not return the element from the underlying range, but
+    instead it's complement
+
+Think about which operator is responsible for returning the element and have special care with the return type
+of that operator as your transformation might make a change necessary.
+\endassignment
+
+\solution
+
+1. The restriction on the alphabet type is done via a `static_assert`:
+\snippet doc/howto/write_a_view/solution_iterator.cpp static_assert
+You could have done this via an additional template constraint, too, but `static_assert` gives you the opportunity
+to give a readable message in case of error.¹
+
+2. The operator that needs to call the `seqan3::complement` function is `operator*`:
+\snippet doc/howto/write_a_view/solution_iterator.cpp dereference
+
+3. As previously noted, care needs to be taken with this function's return type:
+\snippet doc/howto/write_a_view/solution_iterator.cpp reference
+
+
+Here is the full solution:
+\hint
+\include doc/howto/write_a_view/solution_iterator.cpp
+
+The program prints "C T A A T G T "
+
+\endhint
+
+<small>¹ This is only recommended when you do not want to allow a different specialisation of the template to
+cover the excluded case.</small>
+
+\endsolution
+
+You now have a working iterator, although it still lacks the capabilities of `std::BidirectionalIterator`,
+`std::RandomAccessIterator` and `std::ContiguousIterator`.
+When designing views, you should always strive to preserve as much of the capabilities of the underlying range
+as possible.
+
+Which of the mentioned concepts do you think your iterator could be made to implement? Have a look at the respective
+documentation.
```

Comment:
`std::ContiguousIterator` does not link /  not available yet

Reply:
I'll add it to std/ranges